internal/persistence/postgres/sqlpersistence.go

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2023 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"context"
+	"time"
+
+	"github.com/hyperledger/firefly-common/pkg/config"
+	"github.com/hyperledger/firefly-common/pkg/dbsql"
+	"github.com/hyperledger/firefly-common/pkg/ffapi"
+	"github.com/hyperledger/firefly-transaction-manager/internal/persistence"
+	"github.com/hyperledger/firefly-transaction-manager/pkg/apitypes"
+)
+
+const (
+	ConfigTXWriterCacheSlots                = "txwriter.cacheSlots"
+	ConfigTXWriterHistorySummaryLimit       = "txwriter.historySummaryLimit"
+	ConfigTXWriterHistoryCompactionInterval = "txwriter.historyCompactionInterval"
+	ConfigTXWriterCount                     = "txwriter.count"
+	ConfigTXWriterBatchTimeout              = "txwriter.batchTimeout"
+	ConfigTXWriterBatchSize                 = "txwriter.batchSize"
+
+	defaultConnectionLimitPostgreSQL = 50
+)
+
+type sqlPersistence struct {
+	db     *dbsql.Database
+	writer *transactionWriter
+
+	transactions  *dbsql.CrudBase[*apitypes.ManagedTX]
```

Comment:
We have generics now??

Reply:
Yup - the CRUD functionality was the first thing to use them